exonum/src/runtime/types.rs

```diff
@@ -164,14 +180,30 @@ impl ArtifactId {
     ) -> InstanceInitParams {
         InstanceInitParams::new(id, name, self, ())
     }
+}
 
-    /// Checks that the artifact name contains only allowed characters and is not empty.
-    fn is_valid_name(name: impl AsRef<[u8]>) -> bool {
-        // Extended version of `exonum_merkledb::is_valid_name` that also allows '.' and ':'.
-        name.as_ref().iter().all(|&c| match c {
-            b'.' | b':' => true,
-            c => is_allowed_index_name_char(c),
-        })
+impl ProtobufConvert for ArtifactId {
```

Comment:
You can use the convenient protobuf convert attribute `with` to reduce amount of boilerplate code.